api/provider/aws/service_delete.go

```diff
@@ -1,5 +1,65 @@
 package aws
 
-func (s *ServiceProvider) Delete(ServiceID string) error {
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+)
+
+// Delete a specified service within a cluster.
+func (s *ServiceProvider) Delete(serviceID string) error {
+	desiredCount := int64(0)
+
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
+	if err != nil {
+		return err
+	}
+	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
+	fqClusterID := addLayer0Prefix(s.Config.Instance(), environmentID)
+
+	if err := s.scaleService(fqServiceID, fqClusterID, desiredCount); err != nil {
+		return fmt.Errorf("Failed to scale service: %e", err)
+	}
+
+	if err := s.deleteService(fqServiceID, fqClusterID); err != nil {
+		return fmt.Errorf("Failed to delete service: %e", err)
+	}
+
+	if err := deleteEntityTags(s.TagStore, "service", serviceID); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *ServiceProvider) scaleService(serviceID, clusterID string, desiredCount int64) error {
+	input := &ecs.UpdateServiceInput{}
+	input.DesiredCount = &desiredCount
+	input.SetCluster(clusterID)
+	input.SetService(serviceID)
+
+	if err := input.Validate(); err != nil {
+		return fmt.Errorf("Error failed to validate: %e", err)
+	}
+
+	if _, err := s.AWS.ECS.UpdateService(input); err != nil {
+		return fmt.Errorf("Error scaling service: %e", err)
+	}
+	return nil
+}
+
+func (s *ServiceProvider) deleteService(serviceID, clusterID string) error {
+	input := &ecs.DeleteServiceInput{}
+	input.SetCluster(clusterID)
+	input.SetService(serviceID)
+
+	if err := input.Validate(); err != nil {
+		return fmt.Errorf("Error failed to validate: %e", err)
+	}
+
+	if _, err := s.AWS.ECS.DeleteService(input); err != nil {
+		return fmt.Errorf("Error deleting service: %e", err)
```

Comment:
We have a pattern here in other `delete<Entity>` functions where we return the AWS error unless the error indicates that the entity was not found; if we're trying to delete something and we can't find it, mission accomplished anyway because it's not there one way or another.

Here are a couple of examples from `environment_delete.go`:
```
// deleteCluster()
	if _, err := e.AWS.ECS.DeleteCluster(input); err != nil {
		if err, ok := err.(awserr.Error); ok && err.Code() == "ClusterNotFoundException" {
			return nil
		}

		return err
	}
```
```
// deleteLC()
	if _, err := e.AWS.AutoScaling.DeleteLaunchConfiguration(input); err != nil {
		if err, ok := err.(awserr.Error); ok && strings.Contains(err.Message(), "Launch configuration name not found") {
			return nil
		}

		return err
	}
```

Reply:
I noticed pattern when looking at the code and in the current state, the error doesn't get encountered when I did my smoke testing when there isn't a service. 

This is why I decided to include the error. Is there another part of the api that checks if service exists?

Reply:
does it throw an error at all?

Reply:
Oh - if AWS doesn't throw an error when you tell it to delete a service that doesn't exist, then we don't need to look for such an error. Whether we do or don't have logic to check for DNE errors, I think we'd still want to bubble up any other AWS error that might appear.

Reply:
Okay so probably just return an error then, because if an error occurs, it means something else went wrong when attempting a delete, like a task running.